File: clients/algoliasearch-client-php/lib/Model/Personalization/PersonalizationStrategyParams.php
<?php

namespace Algolia\AlgoliaSearch\Model\Personalization;

/**
 * PersonalizationStrategyParams Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class PersonalizationStrategyParams extends \Algolia\AlgoliaSearch\Model\AbstractModel implements ModelInterface, \ArrayAccess, \JsonSerializable
{
    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelTypes = [
        'eventScoring' => '\Algolia\AlgoliaSearch\Model\Personalization\EventScoring[]',
        'facetScoring' => '\Algolia\AlgoliaSearch\Model\Personalization\FacetScoring[]',
        'personalizationImpact' => 'int',
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelFormats = [
        'eventScoring' => null,
        'facetScoring' => null,
        'personalizationImpact' => null,
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelTypes()
    {
        return self::$modelTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelFormats()
    {
        return self::$modelFormats;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'eventScoring' => 'setEventScoring',
        'facetScoring' => 'setFacetScoring',
        'personalizationImpact' => 'setPersonalizationImpact',
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'eventScoring' => 'getEventScoring',
        'facetScoring' => 'getFacetScoring',
        'personalizationImpact' => 'getPersonalizationImpact',
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     */
    public function __construct(array $data = null)
    {
        if (isset($data['eventScoring'])) {
            $this->container['eventScoring'] = $data['eventScoring'];
        }
        if (isset($data['facetScoring'])) {
            $this->container['facetScoring'] = $data['facetScoring'];
        }
        if (isset($data['personalizationImpact'])) {
            $this->container['personalizationImpact'] = $data['personalizationImpact'];
        }
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if (!isset($this->container['eventScoring']) || $this->container['eventScoring'] === null) {
            $invalidProperties[] = "'eventScoring' can't be null";
        }
        if (!isset($this->container['facetScoring']) || $this->container['facetScoring'] === null) {
            $invalidProperties[] = "'facetScoring' can't be null";
        }
        if (!isset($this->container['personalizationImpact']) || $this->container['personalizationImpact'] === null) {
            $invalidProperties[] = "'personalizationImpact' can't be null";
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    /**
     * Gets eventScoring
     *
     * @return \Algolia\AlgoliaSearch\Model\Personalization\EventScoring[]
     */
    public function getEventScoring()
    {
        return $this->container['eventScoring'] ?? null;
    }

    /**
     * Sets eventScoring
     *
     * @param \Algolia\AlgoliaSearch\Model\Personalization\EventScoring[] $eventScoring scores associated with the events
     *
     * @return self
     */
    public function setEventScoring($eventScoring)
    {
        $this->container['eventScoring'] = $eventScoring;

        return $this;
    }

    /**
     * Gets facetScoring
     *
     * @return \Algolia\AlgoliaSearch\Model\Personalization\FacetScoring[]
     */
    public function getFacetScoring()
    {
        return $this->container['facetScoring'] ?? null;
    }

    /**
     * Sets facetScoring
     *
     * @param \Algolia\AlgoliaSearch\Model\Personalization\FacetScoring[] $facetScoring scores associated with the facets
     *
     * @return self
     */
    public function setFacetScoring($facetScoring)
    {
        $this->container['facetScoring'] = $facetScoring;

        return $this;
    }

    /**
     * Gets personalizationImpact
     *
     * @return int
     */
    public function getPersonalizationImpact()
    {
        return $this->container['personalizationImpact'] ?? null;
    }

    /**
     * Sets personalizationImpact
     *
     * @param int $personalizationImpact the impact that personalization has on search results: a number between 0 (personalization disabled) and 100 (personalization fully enabled)
     *
     * @return self
     */
    public function setPersonalizationImpact($personalizationImpact)
    {
        $this->container['personalizationImpact'] = $personalizationImpact;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param int $offset Offset
     *
     * @return bool
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param int $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param int $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/personalization/PersonalizationStrategyParams.java
package com.algolia.model.personalization;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/** PersonalizationStrategyParams */
public class PersonalizationStrategyParams {

  @SerializedName("eventScoring")
  private List<EventScoring> eventScoring = new ArrayList<>();

  @SerializedName("facetScoring")
  private List<FacetScoring> facetScoring = new ArrayList<>();

  @SerializedName("personalizationImpact")
  private Integer personalizationImpact;

  public PersonalizationStrategyParams setEventScoring(
    List<EventScoring> eventScoring
  ) {
    this.eventScoring = eventScoring;
    return this;
  }

  public PersonalizationStrategyParams addEventScoringItem(
    EventScoring eventScoringItem
  ) {
    this.eventScoring.add(eventScoringItem);
    return this;
  }

  /**
   * Scores associated with the events.
   *
   * @return eventScoring
   */
  @javax.annotation.Nonnull
  public List<EventScoring> getEventScoring() {
    return eventScoring;
  }

  public PersonalizationStrategyParams setFacetScoring(
    List<FacetScoring> facetScoring
  ) {
    this.facetScoring = facetScoring;
    return this;
  }

  public PersonalizationStrategyParams addFacetScoringItem(
    FacetScoring facetScoringItem
  ) {
    this.facetScoring.add(facetScoringItem);
    return this;
  }

  /**
   * Scores associated with the facets.
   *
   * @return facetScoring
   */
  @javax.annotation.Nonnull
  public List<FacetScoring> getFacetScoring() {
    return facetScoring;
  }

  public PersonalizationStrategyParams setPersonalizationImpact(
    Integer personalizationImpact
  ) {
    this.personalizationImpact = personalizationImpact;
    return this;
  }

  /**
   * The impact that personalization has on search results: a number between 0 (personalization
   * disabled) and 100 (personalization fully enabled).
   *
   * @return personalizationImpact
   */
  @javax.annotation.Nonnull
  public Integer getPersonalizationImpact() {
    return personalizationImpact;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    PersonalizationStrategyParams personalizationStrategyParams = (PersonalizationStrategyParams) o;
    return (
      Objects.equals(
        this.eventScoring,
        personalizationStrategyParams.eventScoring
      ) &&
      Objects.equals(
        this.facetScoring,
        personalizationStrategyParams.facetScoring
      ) &&
      Objects.equals(
        this.personalizationImpact,
        personalizationStrategyParams.personalizationImpact
      )
    );
  }

  @Override
  public int hashCode() {
    return Objects.hash(eventScoring, facetScoring, personalizationImpact);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class PersonalizationStrategyParams {\n");
    sb
      .append("    eventScoring: ")
      .append(toIndentedString(eventScoring))
      .append("\n");
    sb
      .append("    facetScoring: ")
      .append(toIndentedString(facetScoring))
      .append("\n");
    sb
      .append("    personalizationImpact: ")
      .append(toIndentedString(personalizationImpact))
      .append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
